### PR TITLE
feat!: add `blas/base/strmv`

### PR DESCRIPTION
**Pull request by aman-095**
Progresses: #2039. 

## Description

> What is the purpose of this pull request?

This RFC proposes to add a routine to perform one of the matrix-vector operations `x = A*x`, or `x = A**T*x`, where `x` is an `N` element vector, and `A` is an `N` by `N` unit, or non-unit, upper or lower triangular matrix as defined in [BLAS Level 2](http://www.netlib.org/blas/#_level_2) routines. Specifically adding `@stdlib/blas/base/strmv` is proposed.

## Related Issues

> Does this pull request have any related issues?

This pull request:

- progresses: #2039.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by kgryte** [comment]
@aman-095 Thanks for working on this. As previously discussed, we need to refactor the main and `ndarray` APIs to call a common "base" implementation in order to reduce code duplication. Rather than needing to perform a separate refactoring later, I suggest going ahead and doing that in this PR.

**Comment by kgryte** [comment]
The changes thus far made and the refactoring will also need to be applied to `dtrmv`.

**Comment by aman-095** [comment]
Hey, @kgryte I have applied the required changes here and the similar in #2540.

File: lib/node_modules/@stdlib/blas/base/strmv/lib/base.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+		( isRowMajor && trans !== 'no-transpose' && uplo === 'lower' )
+	) {
+		ix1 = ox;
+		for ( i1 = 0; i1 < N; i1++ ) {
+			if ( x[ ix1 ] !== 0.0 ) {
+				tmp = x[ ix1 ];
+				oa = offsetA + (sa1*i1);
+				ix0 = ox;
+				for ( i0 = 0; i0 < i1; i0++ ) {
+					x[ ix0 ] += f32( tmp * A[ oa+(sa0*i0) ] );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/strmv/lib/base.js]:
```suggestion
					x[ ix0 ] = f32( x[ ix0 ] + f32( tmp * A[ oa+(sa0*i0) ] ) );
```

Comment (kgryte) [reply lib/node_modules/@stdlib/blas/base/strmv/lib/base.js]: We should be consistent in wrapping operations.

Comment (kgryte) [reply lib/node_modules/@stdlib/blas/base/strmv/lib/base.js]: See L100 below.

File: lib/node_modules/@stdlib/blas/base/strmv/lib/base.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+	) {
+		ox += ( N - 1 ) * strideX;
+		ix1 = ox;
+		for ( i1 = N-1; i1 >= 0; i1-- ) {
+			if ( x[ ix1 ] !== 0.0 ) {
+				tmp = x[ ix1 ];
+				oa = offsetA + (sa1*i1);
+				ix0 = ox;
+				for ( i0 = N-1; i0 > i1; i0-- ) {
+					x[ ix0 ] += f32( tmp * A[ oa+(sa0*i0) ] );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/base/strmv/lib/base.js]: Same comment.

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @aman-095!

**Pull request merged**